From b095e4739f6ccc006f6e556d29f4018a3d5129ed Mon Sep 17 00:00:00 2001
From: Peng Liu <liupeng256@huawei.com>
Date: Sat, 21 May 2022 12:18:57 +0800
Subject: [PATCH 2154/2592] [Backport] Documentation: arm64: Add limitation for
 exactmap

hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56F4Z

Reference: https://gitee.com/openeuler/kernel/commit/0d0db072ca083cf5d76e27164f36d976db90b794

--------------------------------

For arm64, the exactmap has to remove all memblock and then add
the user assigned range of memblock. EFI initializes memblock too
late so removing all memblock after it will cause panic. Hence,
the exactmap can be just used in dt boot mode, add this limitation
to Documentation/admin-guide/kernel-parameters.txt.

Signed-off-by: Peng Liu <liupeng256@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
---
 Documentation/admin-guide/kernel-parameters.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Documentation/admin-guide/kernel-parameters.txt b/Documentation/admin-guide/kernel-parameters.txt
index 13db44bae2b7..aa27a059f481 100644
--- a/Documentation/admin-guide/kernel-parameters.txt
+++ b/Documentation/admin-guide/kernel-parameters.txt
@@ -2814,6 +2814,8 @@
 
 	memmap=exactmap	[KNL,X86,ARM64] Enable setting of an exact
 			E820 and ARM64 memory map, as specified by the user.
+			For ARM64, this setting is limited to dt boot mode as
+			exact mapping must be done after initializing memblock.
 			Such memmap=exactmap lines can be constructed based on
 			BIOS output or other requirements. See the memmap=nn@ss
 			option description.
-- 
2.17.1

